widget/app.R

```diff
@@ -0,0 +1,62 @@
+# Copyright(c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+suppressMessages(library(shiny))
+suppressMessages(library(DT))
+
+
+server <- function(input, output, session) {
+  
+  values <- reactiveValues(widget_state = "initializing",
+                           current_run = NULL)
+
+  # retrieve run object once widget_state
+  observeEvent(values$widget_state == "streaming", {
+    ws <- azuremlsdk::get_workspace(ws_name, subscription_id, rg)
+    exp <- azuremlsdk::experiment(ws, exp_name)
+    values$current_run <- azuremlsdk::get_run(exp, run_id)
+  },
+  ignoreInit = TRUE,
+  once = TRUE)
+
+  # stop app if user closes session
+  session$onSessionEnded(function() {
+    shiny::stopApp()
+  })
+  
+  plot <- function() {
+    if (!is.null(values$current_run)) {
+      if (isolate(values$current_run$status) %in% 
+          c("Failed", "Completed", "Canceled")) {
```

Comment:
Can you double check it's "Canceled" or "Cancelled"?

Reply:
I did before writing the code. It's "Canceled".